src/framework/components/anim/state-graph.js

```diff
@@ -1,23 +1,46 @@
 function AnimStateGraph(data) {
     this._layers = [];
     this._parameters = {};
-    if (data) {
-        if (data.states) {
-            this._layers = [];
-            this._layers.push({
-                name: 'Base',
-                states: data.states,
-                transitions: data.transitions
-            });
-        } else {
-            this._layers = data.layers;
-        }
-        this._parameters = {};
-        for (var paramId in data.parameters) {
-            var param = data.parameters[paramId];
-            this._parameters[param.name] = { type: param.type, value: param.value };
+    var i;
+    if (!Array.isArray(data.layers)) {
```

Comment:
Why this test here? Is data.layers sometimes an array? Shouldn't it be testing whether data.layers is an object instead (as the code below assumes)?

Reply:
Yeah, I missed out the assignment of data.layers in the case it's an array. I've added this in now.